allspice/utils/bom_generation.py

```diff
@@ -314,22 +383,24 @@ def _group_entries(
 
     for components in grouped_components.values():
         row = {}
-        for column in group_by:
-            # The RHS here shouldn't fail as we've validated the group by
-            # columns are all in the column selection.
-            row[column] = components[0][column]
-        non_group_by = set(components[0].keys()) - set(group_by)
-        for column in non_group_by:
-            # For each of the values in the non-group-by columns, we take the
-            # unique values from all the components and join them with a comma.
-            # This is better than taking the non-unique values and joining them
-            # with a comma, because it means a user wouldn't have to group by
-            # more columns than they want to.
-            row[column] = ", ".join(
-                # dict.fromkeys retains the insertion order; set doesn't.
-                dict.fromkeys(str(component[column]) for component in components).keys()
-            )
-        row["Quantity"] = str(len(components))
+
+        for column, column_config in columns_mapping.items():
+            if column in group_by:
+                # The RHS here shouldn't fail as we've validated the group by
+                # columns are all in the column selection.
+                row[column] = components[0][column]
+            else:
```

Comment:
So if we don't group by the column we display a list of all the column values (maybe deduplicated).

Commenting the function to help break up the flow and understand each part would help here.

Reply:
Added comments to explain the flow here, let me know if this works or I should go into more detail.